keras_cv/layers/preprocessing/random_cutout.py

```diff
@@ -147,46 +113,37 @@ def augment_image(self, image, transformation=None):
 
     def _compute_rectangle_position(self, inputs):
         input_shape = tf.shape(inputs)
-        batch_size, image_height, image_width = (
+        image_height, image_width = (
             input_shape[0],
             input_shape[1],
-            input_shape[2],
         )
         center_x = self._random_generator.random_uniform(
-            [batch_size], 0, image_width, dtype=tf.int32
+            [1], 0, image_width, dtype=tf.int32
         )
         center_y = self._random_generator.random_uniform(
-            [batch_size], 0, image_height, dtype=tf.int32
+            [1], 0, image_height, dtype=tf.int32
         )
         return center_x, center_y
 
     def _compute_rectangle_size(self, inputs):
         input_shape = tf.shape(inputs)
-        batch_size, image_height, image_width = (
+        image_height, image_width = (
             input_shape[0],
             input_shape[1],
-            input_shape[2],
         )
-        height = self._random_generator.random_uniform(
-            [batch_size], self.height_lower, self.height_upper, dtype=tf.float32
-        )
-        width = self._random_generator.random_uniform(
-            [batch_size], self.width_lower, self.width_upper, dtype=tf.float32
-        )
-
-        if self._height_is_float:
-            height = height * tf.cast(image_height, tf.float32)
+        height = self.height_factor()
```

Comment:
This was somehow confusing, since I was trying to find the method height_factor in this class, but the height_factor is an callable attribute.

Reply:
Interesting - @fchollet recommended having a call method to match `self.optimizer()`, etc.  

I think it is more consistent with the Keras API to use __call__.  I think the lack of the method definition is mildly confusing, but with ctrl-F, a first time ready will quickly find the definition which should help.

Thoughts?  I have never been a first time reader of this code so definitely open to discussion.

Reply:
I think when the __call__ method get some inputs, then it will make more sense to read. Most of our __call__ methods do have inputs, which is why I feel this one is bit weird to read. Do we expect any inputs to the call?